pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -213,7 +213,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(
 		VirtualHosts:                   virtualHosts,
 		ValidateClusters:               proto.BoolFalse,
 		MaxDirectResponseBodySizeBytes: istio_route.DefaultMaxDirectResponseBodySizeBytes,
-		IgnorePortInHostMatching:       true,
+		IgnorePortInHostMatching:       IgnorePortInHostMatching(listenerPort, node),
```

Comment:
It was intended to ignore port during host match for sidecar

Reply:
Not always. When listenerPort is 0, it will generate virtual host's domains with port, and now if sidecar ignore port during host matching, it can not match

Reply:
That sounds like a bug in the generation, not here?

Can you give more info on the exact setup that this fixes (and maybe add a test)?

Reply:
I kind of understand this bug. There is a inconsistency when listenerport = 0.

Reply:
> That sounds like a bug in the generation, not here?
>
> Can you give more info on the exact setup that this fixes (and maybe add a test)?

For example, when listenerPort = 0, sidecar will get dynamic route configs like
```
"dynamic_route_configs": [
{
  "name": "0",`
  "virtual_hosts": [
    "name": "a-b-c.svc.cluster.local:10000",
    "domains": [
        "a-b-c.svc.cluster.local:10000",
        "a-b-c:10000",
    ]
  ]
}]
```
When listenerPort = 10000, sidecar will get dynamic route configs like
```
"dynamic_route_configs": [
{
  "name": "10000",`
  "virtual_hosts": [
    "name": "a-b-c.svc.cluster.local:10000",
    "domains": [
        "a-b-c.svc.cluster.local",
        "a-b-c",
    ]
  ]
}]
```
If we set IgnorePortInHostMatching true, sidecar will match domains with "a-b-c.svc.cluster.local", not "a-b-c.svc.cluster.local:10000". It can not match when listenerPort = 0.
Further more, when listenerPort = 0 we could not generate domain "a-b-c.svc.cluster.local" and "a-b-c" in buildVirtualHost step because service a-b-c may has more than one port.